File: src/haveapi/validators/length.js
Validator.validators.length = function (opts, value) {
	if (typeof value != 'string')
		return false;

	var len = value.length;

	if (typeof opts.equals === 'number')
		return len === opts.equals;

	if (typeof opts.min === 'number' && !(typeof opts.max === 'number'))
		return len >= opts.min;

	if (!(typeof opts.min === 'number') && typeof opts.max === 'number')
		return len <= opts.max;

	return len >= opts.min && len <= opts.max;
};

File: clients/ruby/lib/haveapi/client/validators/length.rb
require 'haveapi/client/validator'

module HaveAPI::Client
  class Validators::Length < Validator
    name :length

    def valid?
      len = value.length

      return len == opts[:equals] if opts[:equals]
      return len >= opts[:min] if opts[:min] && !opts[:max]
      return len <= opts[:max] if !opts[:min] && opts[:max]
      len >= opts[:min] && len <= opts[:max]
    end
  end
end
